dlrover/python/elastic_agent/monitor/resource.py

```diff
@@ -92,32 +92,45 @@ def __init__(self):
         reports the used memory and cpu percent to the DLRover master.
         """
         self._total_cpu = psutil.cpu_count(logical=True)
-        self.init_gpu_monitor()
-        logger.info("Resource Monitor Initializing ...")
+        self._gpu_enabled = False
+        self._gpu_stats: list[GPUMetric] = []
+
+    def start(self):
         if (
             os.getenv(NodeEnv.DLROVER_MASTER_ADDR, "")
             and os.getenv(NodeEnv.AUTO_MONITOR_WORKLOAD, "") == "true"
         ):
-            try:
-                thread = threading.Thread(
-                    target=self._monitor_resource,
-                    name="monitor_resource",
-                    daemon=True,
-                )
-                thread.start()
-                if thread.is_alive():
-                    logger.info("Resource Monitor initialized successfully")
-            except Exception as e:
-                logger.error(
-                    f"Failed to start the monitor resource thread. Error: {e}"
-                )
+            return
```

Comment:
Why we should return here when `MASTER_ADDR` & `AUTO_MONITOR` is `True`?

Reply:
This is a bug and I have fixed it.